crates/modules/process-monitor/src/lib.rs

```diff
@@ -150,22 +147,40 @@ pub mod pulsar {
             BpfSenderWrapper::new(ctx.get_sender(), move |event: &BpfEvent<ProcessEvent>| {
                 let _ = tx_processes.send(match event.payload {
                     ProcessEvent::Fork {
+                        uid,
                         ppid,
                         namespaces,
-                        is_new_container,
-                    } => TrackerUpdate::Fork {
-                        pid: event.pid,
-                        ppid,
-                        timestamp: event.timestamp,
-                        namespaces,
-                        is_new_container,
-                    },
+                        ref c_container_id,
+                    } => {
+                        let container_id = match c_container_id {
+                            COption::Some(ccid) => {
+                                let id = ccid.cgroup_id.string(&event.buffer).unwrap();
+                                let container_id = match ccid.container_engine {
+                                    0 => ContainerId::Docker(id),
+                                    1 => ContainerId::Libpod(id),
+                                    _ => panic!("unknown container id identifier"),
```

Comment:
Could we rather log an error or warning and just return `None` here? Panicking is probably not the best idea (we are potentially losing an entire event only because container info was malformed).

Reply:
Not sure if we have to proceed with a `None`. Let me explain:

This piece of code is responsible to the FFI structure conversion: its purpose is to pass from a map buffer allocated string to a heap allocated string. Imagine if the dynamic memory was not involved and it was just an FFI struct with all fields on the stack, you wouldn't care about checking the content of the struct byte by byte to ensure there is a valid discriminant for a any enum field, but you simply interpreted the raw data as a FFI type.
In general if you make this kind of mistake, you end with an undefined behaviour.  

Probably the `panic!` doesn't communicate intent and I should use `unreachable!`. This is not an unrecoverable error and neither you can ignore it, it's a developer fault

Reply:
or maybe we should use the even extreme `unsafe` function `std::hint::unreachable_unchecked()`. 

Basically this is a value invariant. similar to `bool` type with only 2 possible values

Reply:
After the last change there is no need to handle corner cases if we use this enum on the FFI type:

```rust
#[repr(C)]
pub enum ContainerEngineKind {
    Docker,
    Podman,
}
```

Reply:
Nice solution with the last change!

Thanks for explanation! I would be fine with `unreachable` too.